lib/rubocop/cop/lint/uri_regexp.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop identifies places where `URI.regexp`
+      # can be replaced by `URI::DEFAULT_PARSER.make_regexp`.
```

Comment:
I think you should also mention that `URI.regexp` was deprecated and this is the reason why the cop exists. 

Right now it's definitely not clear why this is being flagged.

Reply:
I improved to the offense message that `URI.regexp` is obsolete. 00835fb
What do you think about this?